mico-core/src/main/java/io/github/ust/mico/core/persistence/MicoServiceDeploymentInfoRepository.java

```diff
@@ -135,30 +134,33 @@
      * Deletes all deployment information for all versions of an application including the ones for the KafkaFaasConnectors.
      * All additional properties of a {@link MicoServiceDeploymentInfo} that
      * are stored as a separate node entity and connected to it via a {@code [:HAS]}
-     * relationship will be deleted, too.
+     * relationship will be deleted too, if they are used exclusively by this deployment information.
      *
      * @param applicationShortName the short name of the {@link MicoApplication}.
      */
     @Query("MATCH (a:MicoApplication)-[:PROVIDES|:PROVIDES_KF_CONNECTOR]->(sdi:MicoServiceDeploymentInfo)-[:FOR]->(s:MicoService) "
         + "WHERE a.shortName = {applicationShortName} "
-        + "WITH a, sdi OPTIONAL MATCH (sdi)-[:HAS]->(additionalProperty) "
-        + "WHERE a.shortName = {applicationShortName} "
-        + "DETACH DELETE sdi, additionalProperty")
+        + "WITH a, sdi "
+        + "OPTIONAL MATCH (sdi)-[:HAS]->(allRelatedNodes) "
+        + "WHERE size((allRelatedNodes)--()) = 1 "
```

Comment:
Maybe leave a comment to explain this line: "Only one edge means this relatedNode has only an edge to the sdi we want to delete. Therefor we can also delete this related node safely."